src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/UpdateWorkitemProcedureStepState.sql

```diff
@@ -100,8 +101,7 @@ BEGIN
 			        wi.PartitionKey = eqts.PartitionKey
 			        AND wi.WorkitemKey = eqts.SopInstanceKey1
 	        WHERE
-		        wi.PartitionKey = @partitionKey
-		        AND wi.WorkitemKey = @workitemKey
+		        wi.WorkitemKey = @workitemKey
         )
```

Comment:
this should be proposedWatermark

Reply:
Or we should flip the logic, first update the tag and then update the watermark since they are all in transaction

Reply:
We don't need watermark in this where clause. removed it.

Reply:
Updated the script